tripal_chado/src/Plugin/Field/FieldType/ChadoPropertyTypeDefault.php

```diff
@@ -232,4 +226,84 @@ public function isCompatible(TripalEntityType $entity_type) : bool {
     return $compatible;
   }
 
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\tripal\TripalField\Interfaces\TripalFieldItemInterface::discover()
+   */
+  public static function discover(TripalEntityType $bundle, string $field_id, array $field_definitions) : array{
+
+    /** @var \Drupal\tripal_chado\Database\ChadoConnection $chado **/
+    $chado = \Drupal::service('tripal_chado.database');
+
+    // Initialize with an empty field list.
+    $field_list = [];
+
+    // Make sure the base table setting exists.
+    $base_table = $bundle->getThirdPartySetting('tripal', 'chado_base_table');
+    if (!$base_table) {
+      return $field_list;
+    }
+
+    // Make sure the prop table exists in Chado.
+    $prop_table = $base_table . 'prop';
+    if (!$chado->schema()->tableExists($prop_table)) {
+      return $field_list;
+    }
+
+    // Search for all unique types in the prop table.
+    $query = $chado->select('1:' . $prop_table, 'pt');
+    $query->leftJoin('1:cvterm', 'cvt', 'pt.type_id = cvt.cvterm_id');
+    $query->leftJoin('1:dbxref', 'dbx', 'dbx.dbxref_id = cvt.dbxref_id');
+    $query->leftJoin('1:db', 'db', 'db.db_id = dbx.db_id');
+    $query->leftJoin('1:cv', 'cv', 'cv.cv_id = cvt.cv_id');
+    $query->addField('cvt', 'cvterm_id');
+    $query->addField('cvt', 'name', 'cvterm_name');
+    $query->addField('cvt', 'definition');
+    $query->addField('dbx', 'accession');
+    $query->addField('db', 'name', 'db_name');
+    $query->addField('cv', 'name', 'cv_name');
+    $results = $query->distinct()->execute()->fetchAll();
+
+    // Create a field entry for each property type.
+    foreach ($results as $recprop) {
+      $field_list[] = [
+        'name' => self::generateFieldName($bundle, $recprop->cvterm_name),
+        'content_type' => $bundle->getID(),
+        'label' => ucwords($recprop->cvterm_name),
+        'type' => self::$id,
+        'description' => 'A record property with the following definition: ' . $recprop->definition,
```

Comment:
Here is one last comment for now. I tested importing a different GFF file and I get all of the properties, but in every case there is no definition following this text `A record property with the following definition: `

![2024-05-22_dcar](https://github.com/tripal/tripal/assets/8419404/10b70ec5-bc4f-4a19-a0bb-9036b2fb7165)

Well that is because there is no definition. I wonder if the fallback should be to use the term when there is no definition? Or always include the namespace and term anyway?

Reply:
Yah, I agree this description needs help. Maybe the pattern should be:
- if the definition is empty: "Chado Properties tagged with the term $term_name ($term_accession)."
- if the definition is available: "Chado Properties tagged with the term $term_name ($term_accession) which is defined as '$term_definition'."

Reply:
Fixed in most recent commit.